requre/postprocessing.py

```diff
@@ -112,3 +121,113 @@ def simplify(
             else:
                 for v in internal_object.values():
                     self.simplify(internal_object=v, ignore_list=ignore_list)
+
+
+class tarFilesSimilarity:
+    def __init__(self, path, hash_function=None):
+        self.mapping_table = {}
+        self.path = path
+        for tar_file in self.list_files(path):
+            current_hash = self.untar_and_return_hash(tar_file, hash_function)
+            self.mapping_table[tar_file] = current_hash
```

Comment:
It's usually not a good approach to do a lot of work in the `__init__` method, can we move that to some method and call outside of the `__init__`?

Reply:
You can create `mapping_table` as a cached property.

(You can use [`lru_cache`/`cached_property`](https://github.com/packit/ogr/blob/master/ogr/services/base.py#L45) if you don't want to implement that.)

Reply:
that's true, that this mapping table could be reated on demand, not in init.